content/200-concepts/100-components/03-prisma-migrate/010-mental-model.mdx

```diff
@@ -0,0 +1,178 @@
+
+## Evolve your database schema with Prisma Migrate
+
+This section describes how you can evolve your database schema in different environments: development, staging, and production, using Prisma Migrate.
+
+### Prisma Migrate in a development environment (local)
+
+#### Track your migration history with `prisma migrate dev`
+
+The [`prisma migrate dev`](/reference/api-reference/command-reference#migrate-dev) command allows you to track the changes you make to your database. The `prisma migrate dev` command automatically generates SQL migration files (saved in `/prisma/migrations`) and applies them to the database. When a migration is applied to the database, the migrations table (`_prisma_migrations`) in your database is also updated.
+
+![Prisma Migrate dev flow](https://user-images.githubusercontent.com/33921841/189875756-5df6ba94-c7a9-4e9c-a8f7-ce4d5d09b204.png)
+
+The `prisma migrate dev` command tracks the state of the database using the following pieces of state:
+
+- the Prisma schema
+- the migrations history
+- the migrations table
+- the database schema
+
+> **Note**: The pieces of state used to track the state of a migration are the same as the ones described in [how Prisma Migrate tracks the migration state](#how-prisma-migrate-tracks-the-migration-state) section.
+
+You can customize migrations before you apply them to the database using the `--create-only` flag. For example, you might want to edit a migration if you want to rename columns without incurring any data loss or load database extensions (in PostgreSQL) and database views (currently not supported).
+
+Under the hood, Prisma Migrate uses a [shadow database](/concepts/components/prisma-migrate/shadow-database) to detect a [schema drift](/concepts/components/prisma-migrate/shadow-database#detecting-schema-drift) and generate new migrations.
+
+> **Note**: `prisma migrate dev` is intended to be used only in development with a disposable database.
+
+If `prisma migrate dev` detects a schema drift or a migration history conflict, you will be prompted to reset (drop and recreate your database) your database to sync the migration history and the database schema.
+
+<details>
+
+<summary> Expand to see the shadow database explained using a cartoon</summary>
+
+![A cartoon that shows how the shadow database works.](shadow-database.png)
+
+</details>
+
+#### Resolve schema drifts
+
+A schema drift occurs when the expected database schema is different from what is in the migration history. For example, this can occur when you manually update the database schema without also updating the Prisma schema and `prisma/migrations` accordingly.
+
+For such instances, you can use the [`prisma migrate diff`](/reference/api-reference/command-reference#migrate-diff) command to compare your migration history and revert changes made to your database schema.
```

Comment:
In some situations, it might not be ideal to reset development environments. `migrate diff` can help understand the drift, and perhaps update the migration being applied to the database.